debug/ftrace.rst

```diff
@@ -36,3 +36,42 @@ Usage
     - Run helper script called ``symbolize.py`` to translate the function graph
       addresses into function names: ``cat ftrace-<ta_uuid>.out |
       ./optee_os/scripts/symbolize.py -d <ta_uuid>.elf``
+
+Typical output
+**************
+
+.. code-block:: bash
+
+  # tracer: function_graph
+  #
+  # DURATION          FUNCTION CALLS
+  #  |   |             |   |   |   |
+
+              | __ta_entry() {
+              |  __utee_entry() {
+     1.664 us |   ta_header_get_session();
+    11.264 us |   from_utee_params();
+      .896 us |   memcpy();
+              |   TA_InvokeCommandEntryPoint() {
+              |    TEE_GenerateRandom() {
+   163.360 us |     utee_cryp_random_number_generate();
+   186.848 us |    }
+   214.288 us |   }
+    19.088 us |   to_utee_params();
+              |   ta_header_save_params() {
+      .736 us |    memset();
+     2.832 us |   }
+   304.880 us |  }
+   307.168 us | }
+
+
+The duration (function's time of execution) is displayed on the closing bracket
+line of a function or on the same line in case the function is the leaf one.
+IOW, duration is displayed whenever an instrumented function returns. For user
+space functions, it comprises the time spent executing user space code in the
+function and any of its callees (OP-TEE CPU time [1]), plus the time spent
+executing or waiting for any system calls to proceed (OP-TEE CPU time [1]). For
+kernel space function, it is the OP-TEE CPU time [1] between function entry and
+exit.
+
+[1] OP-TEE CPU time: CPU execution time allocated by REE OS scheduler.
```

Comment:
I find this unclear and possibly wrong (waiting time... could be interrupted so part of it could be subtracted etc.). How about saying what is actually implemented? Replace everything starting with "For user space functions" with:

"*It comprises the time spent executing the function and any of its callees. The Counter-timer Physical Count register (CNTPCT) and the Counter-timer Frequency register (CNTFRQ) are used to compute durations. Time spent in foreign interrupts is subtracted.*"

Reply:
Okay will instead use this text instead.